qiskit_experiments/quantum_volume/qv_experiment.py

```diff
@@ -0,0 +1,129 @@
+from qiskit.providers.backend import Backend
+from qiskit.providers.options import Options
+
+try:
+    from qiskit import Aer
+
+    HAS_SIMULATION_BACKEND = True
+except ImportError:
+    HAS_SIMULATION_BACKEND = False
+
+from qiskit import QuantumCircuit
+from qiskit.circuit.library import QuantumVolume as QuantumVolumeCircuit
+from qiskit import execute
+from qiskit_experiments.base_experiment import BaseExperiment
+from .qv_analysis import QuantumVolumeAnalysis
+
+
+class QuantumVolume(BaseExperiment):
+    """Quantum Volume Experiment class
+
+    Experiment Options:
+        trials (int): Optional, number of times to generate new Quantum Volume circuits and
+                    calculate their heavy output.
+    """
+
+    # Analysis class for experiment
+    __analysis_class__ = QuantumVolumeAnalysis
+
+    def __init__(
+        self,
+        qubits: Union[int, Iterable[int]],
+        trials: Optional[int] = 100,
+        seed: Optional[Union[int, Generator]] = None,
+        simulation_backend: Optional[Backend] = None,
+    ):
+        """Quantum Volume experiment
+        Args:
+            qubits: the number of qubits or list of
+                    physical qubits for the experiment.
+            trials: number of trials to run the quantum volume circuit.
+            seed: Seed or generator object for random number
+                  generation. If None default_rng will be used.
+            simulation_backend: the simulator backend to use to generate
+                the expected results. the simulator must have a 'save_probabilities' method.
+                if None Aer simulator will be used (in case Aer is not installed -
+                qiskit.quantum_info will be used).
+        """
+        super().__init__(qubits)
+
+        # Set configurable options
+        self.set_experiment_options(trials=trials)
+
+        # Set fixed options
+        self._previous_trials = 0
+
+        if not isinstance(seed, Generator):
+            self._rng = default_rng(seed=seed)
+        else:
+            self._rng = seed
+
+        if not simulation_backend and HAS_SIMULATION_BACKEND:
+            self._simulation_backend = Aer.get_backend("aer_simulator")
+        else:
+            self._simulation_backend = simulation_backend
+
+    @classmethod
+    def _default_experiment_options(cls):
+        return Options(trials=100)
+
+    def _get_ideal_data(self, circuit, **run_options):
+        """
+        In case the user does not have Aer installed - use Terra to calculate the ideal state
+        Args:
+            circuit: the circuit to extract the ideal data from
+        Returns:
+            dict: the probability for each state in the circuit
+        """
+        ideal_circuit = circuit.remove_final_measurements(inplace=False)
+        ideal_circuit.save_probabilities()
+        if self._simulation_backend:
+            ideal_result = execute(ideal_circuit, backend=self._simulation_backend, **run_options)
```

Comment:
This should be changed to avoid execute:
```python
ideal_circuit = transpile(ideal_circuit, self._simulation_backend)
ideal_result = self._simulation_backend.run(ideal_circuit, **run_options).result()
```
To avoid transpiling twice we could also move this functionality of adding the target probs into the `_postprocess_transpiled_circuits` method, then it would just been to call `backend.run`

Reply:
We don't want to run simulation on the transpiled circuits, if I recall our discussion in slack, so I don't think we can use `_postprocess_transpiled_circuits` here. We really need to transpile two different circuits.

Reply:
I agree. We should simulate the ideal circuits without transpiling them, either by the `Statevector` class or using Aer simulator.

Reply:
I agree with Shelly and Gadi. I removed the deprecated execute method and replaced it with transpilation, but one which is without the parameters from the user, so the optimization_level should stay 0, so as far as I understand, the circuits should stay the same (because we want to compare to the probabilities of the exact generated circuits without any approximations)

Reply:
Perhaps it's worth to provide explicitly the parameter `optimization_level=0` (I think that the default is 1), and add a comment about this in the code.
I am afraid that as part of the transpilation, the unitary might be slightly changed to reduce the number of gates, but this should not happen in `optimization_level=0`.